Modules/Challenges/test/SpecFlowChallenges.Specs/API/ChallengesApi.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Net;
+using RestSharp;
+using SpecFlowChallenges.Specs.Models;
+
+namespace SpecFlowChallenges.Specs.API;
+public class ChallengesApi
+{
+    private readonly RestClient _client;
+
+    public ChallengesApi(RestClient client)
+    {
+        _client = client;
+
+        ServicePointManager.ServerCertificateValidationCallback +=
+                (sender, cert, chain, sslPolicyErrors) => true;
+    }
+
+    public async Task<RestResponse<AccessTokenResponse>> GetAccessTokenAsync(AuthenticationParameters authenticationParams)
+    {
+        var request = new RestRequest("/connect/token", Method.Post);
+
+        //RestSharp will use the correct content type by default (https://restsharp.dev/usage.html#http-header)
+        //request.AddHeader("Content-Type", "application/x-www-form-urlencoded");
+
+        foreach (var parameter in authenticationParams.Parameters)
+        {
+            request.AddParameter(parameter.Key, parameter.Value);
+        }
+
+        return await _client.ExecuteAsync<AccessTokenResponse>(request);
+    }
+
+    public async Task<RestResponse<ChallengeResponse>> CreateChallengeAsync(RequestConfiguration requestConfiguration)
```

Comment:
You shouldn't return RestResponse here, in order to stay independent of the used rest client.